back/app/services/permissions/base_permissions_service.rb

```diff
@@ -66,41 +66,30 @@ def find_permission(action, phase)
 
     # User methods
     def user_denied_reason(permission, user, scope = nil)
-      if permission.permitted_by == 'everyone'
-        user ||= User.new
-      else
-        return USER_DENIED_REASONS[:user_not_signed_in] unless user
-        return USER_DENIED_REASONS[:user_blocked] if user.blocked?
-
-        unless user.confirmation_required? # Ignore non confirmed users as this will be picked up by UserRequirementsService
-          return USER_DENIED_REASONS[:user_not_active] unless user.active?
-          return if UserRoleService.new.can_moderate? scope, user
-          return USER_DENIED_REASONS[:user_not_permitted] if permission.permitted_by == 'admins_moderators'
-
-          if permission.permitted_by == 'groups'
-            reason = denied_when_permitted_by_groups?(permission, user)
-            return USER_DENIED_REASONS[reason] if reason.present?
-          end
-        end
-      end
-      return if user_requirements_service.requirements(permission, user)[:permitted]
-
-      USER_DENIED_REASONS[:user_missing_requirements]
+      return if permission.permitted_by == 'everyone'
+      return USER_DENIED_REASONS[:user_not_signed_in] unless user
+      return USER_DENIED_REASONS[:user_blocked] if user.blocked?
+      return USER_DENIED_REASONS[:user_missing_requirements] if user.confirmation_required?
+      return USER_DENIED_REASONS[:user_not_active] unless user.active?
+      return if UserRoleService.new.can_moderate? scope, user
+      return USER_DENIED_REASONS[:user_not_permitted] if permission.permitted_by == 'admins_moderators'
+      return USER_DENIED_REASONS[:user_missing_requirements] unless user_requirements_service.requirements(permission, user)[:permitted]
+      return USER_DENIED_REASONS[:user_not_verified] if user_requirements_service.requires_verification?(permission, user)
+      return USER_DENIED_REASONS[:user_not_in_group] if denied_when_permitted_by_groups?(permission, user)
+
+      nil
```

Comment:
It's great you got rid of the nesting. Previously, it reminded me a bit this infamous kata :smile: https://github.com/emilybache/GildedRose-Refactoring-Kata/blob/main/ruby/gilded_rose.rb

Though I think it's totally fine as it is, maybe this other structure makes reading conditions a bit easier. Or maybe not :thinking: Up to you.

```suggestion
      if permission.permitted_by == 'everyone'
        nil
      elsif !user
        USER_DENIED_REASONS[:user_not_signed_in]
      elsif user.blocked?
        USER_DENIED_REASONS[:user_blocked]
      elsif user.confirmation_required?
        USER_DENIED_REASONS[:user_missing_requirements]
      elsif !user.active?
        USER_DENIED_REASONS[:user_not_active]
      elsif UserRoleService.new.can_moderate?(scope, user)
        nil
      elsif permission.permitted_by == 'admins_moderators'
        USER_DENIED_REASONS[:user_not_permitted]
      elsif !user_requirements_service.requirements(permission, user)[:permitted]
        USER_DENIED_REASONS[:user_missing_requirements]
      elsif user_requirements_service.requires_verification?(permission, user)
        USER_DENIED_REASONS[:user_not_verified]
      elsif denied_when_permitted_by_groups?(permission, user)
        USER_DENIED_REASONS[:user_not_in_group]
      else
        nil
      end
```

Reply:
Think it may well be personal preference here - I'm not a huge fan of big nested `elsif`s tbh